docs/control-structures.rst

```diff
@@ -276,12 +276,7 @@ Ranges are created using the ``range`` function. The following examples are vali
     for i in range(stop, bound=N):
         ...
 
-``stop`` can be any integer greater than zero. ``N`` must be a compile-time constant. ``i`` begins as zero and increments by one until it is equal to ``stop``. If ``stop`` is larger than ``N``, execution will revert at runtime. To avoid the possibility of runtime reversion, use this in combination with ``min()``:
-
-.. code-block:: python
-
-    for i in range(min(stop, N), bound=N):
-        ...
+``stop`` can be any integer greater than zero. ``N`` must be a compile-time constant. ``i`` begins as zero and increments by one until it is equal to ``stop``. If ``stop`` is larger than ``N``, execution will revert at runtime. In certain cases, you may not have a guarantee that ``stop`` is less than ``N``, but still want to avoid the possibility of runtime reversion. To accomplish this, use the ``bound=`` keyword in combination with ``min()``, like ``range(min(stop, N), bound=N)``.
```

Comment:
If it's zero will it raise? Or avoid the loop?

Reply:
```suggestion
``stop`` can be any integer greater than zero. ``N`` must be a compile-time constant. ``i`` begins as zero and increments by one until it is equal to ``stop``. If ``stop`` is larger than ``N``, execution will revert at runtime. In certain cases, you may not have a guarantee that ``stop`` is less than ``N``, but still want to avoid the possibility of runtime reversion. To accomplish this, use the ``bound=`` keyword in combination with ``min(stop, N)`` as the argument to ``range``, like ``range(min(stop, N), bound=N)``. This is helpful for use cases like chunking up operations on larger arrays across multiple transactions.
```

Reply:
> If it's zero will it raise? Or avoid the loop?

Sorry @charles-cooper, made two comments at once here lol

Reply:
> If it's zero will it raise? Or avoid the loop?

it will skip the loop. again, this is clear in the tests from the implementing PR https://github.com/vyperlang/vyper/commit/d48438e10722db8fc9e74d8ed434745e3b0d31cf#diff-c3f9a43de8dca93393504fef8b3c6bd8c77a6be969ee749f59b49537285e6392R27 but i've updated it to be just a little more clear in https://github.com/vyperlang/vyper/pull/3540/commits/b613cb98f455c426d4dc00b448ec3b0a046a8397

Reply:
Can add that note to the docs too?